crates/uv/src/commands/pip/operations.rs

```diff
@@ -60,8 +61,18 @@ pub(crate) async fn read_requirements(
     // If the user requests `extras` but does not provide a valid source (e.g., a `pyproject.toml`),
     // return an error.
     if !extras.is_empty() && !requirements.iter().any(RequirementsSource::allows_extras) {
+        let hint = if requirements.iter().any(|source| {
+            matches!(
+                source,
+                RequirementsSource::Editable(_) | RequirementsSource::SourceTree(_)
+            )
+        }) {
+            "Use `<dir>[extra]` syntax or `-r <file>` instead."
+        } else {
+            "Use `package[extra]` syntax instead."
```

Comment:
Not in love, but... what am I to do?

Reply:
Wait really, that works but these alt flags don't?

Reply:
Yeah...

So.. `pip install package[extra]` is what's supported upstream. Downstream, we added support for `pip install -r pyproject.toml` but there's no syntax to select extras there so we have `--extra` and `--all-extras`. These are also copied over from `uv pip compile pyproject.toml` where they make a little more sense. `uv pip compile .` doesn't exist.